src/main/java/javaslang/collection/Vector.java

```diff
@@ -1180,6 +1191,23 @@ public int length() {
         }
         return Tuple.of(xs.isEmpty() ? empty() : new Vector<>(xs), ys.isEmpty() ? empty() : new Vector<>(ys));
     }
+    
+    @Override
+    public <T1, T2, T3> Tuple3<Vector<T1>, Vector<T2>, Vector<T3>> unzip3(Function<? super T, Tuple3<? extends T1, ? extends T2, ? extends T3>> unzipper) {
+        Objects.requireNonNull(unzipper, "unzipper is null");
+        HashArrayMappedTrie<Integer, T1> xs = HashArrayMappedTrie.empty();
+        HashArrayMappedTrie<Integer, T2> ys = HashArrayMappedTrie.empty();
+		HashArrayMappedTrie<Integer, T3> zs = HashArrayMappedTrie.empty();
+        for (T element : this) {
+            final Tuple3<? extends T1, ? extends T2, ? extends T3> t = unzipper.apply(element);
+            xs = xs.put(xs.size(), t._1);
+            ys = ys.put(ys.size(), t._2);
+			zs = zs.put(zs.size(), t._3);
+        }
+        return Tuple.of(xs.isEmpty() ? empty() : new Vector<>(xs), 
+						ys.isEmpty() ? empty() : new Vector<>(ys),
+						zs.isEmpty() ? empty() : new Vector<>(zs));
```

Comment:
Yeah it isn't, it comes from the unzip method. That's what you get copying other people's code. :)